### PR TITLE
Fix outdated `verification_mode` values

### PR DESCRIPTION
**Pull request by polinaeterna**
~I think it makes sense not to save `dataset_info.json` file to a dataset cache directory when loading dataset with `verification_mode="no_checks"` because otherwise when next time the dataset is loaded **without** `verification_mode="no_checks"`, it will be loaded successfully, despite some values in info might not correspond to the ones in the repo which was the reason for using `verification_mode="no_checks"` first.~

Updated values of `verification_mode` to the current ones in some places ("none" -> "no_checks", "all" -> "all_checks")

### CODE REVIEW

File: src/datasets/builder.py [hunk 1]
```diff
@@ -357,10 +357,11 @@ def __init__(
             os.makedirs(self._cache_dir_root, exist_ok=True)
             lock_path = os.path.join(self._cache_dir_root, self._cache_dir.replace(os.sep, "_") + ".lock")
             with FileLock(lock_path):
-                if os.path.exists(self._cache_dir):  # check if data exist
-                    if len(os.listdir(self._cache_dir)) > 0:
-                        logger.info("Overwrite dataset info from restored data version.")
-                        self.info = DatasetInfo.from_directory(self._cache_dir)
+                if os.path.exists(self._cache_dir) > 0:  # check if data exist
+                    if len(os.listdir(self._cache_dir)):
```
Comment (lhoestq) [inline src/datasets/builder.py:361]:
```suggestion
                if os.path.exists(self._cache_dir):  # check if data exist
                    if len(os.listdir(self._cache_dir)) > 0:
```

File: src/datasets/builder.py [hunk 2]
```diff
@@ -880,7 +881,8 @@ def incomplete_dir(dirname):
                     self.info.download_checksums = dl_manager.get_recorded_sizes_checksums()
                     self.info.size_in_bytes = self.info.dataset_size + self.info.download_size
                     # Save info
-                    self._save_info()
+                    if verification_mode is not VerificationMode.NO_CHECKS:
+                        self._save_info()
```
Comment (lhoestq) [inline src/datasets/builder.py:885]: We should always save the info. For example it stores the feature types and the arrow shards lengths.

**Review by lhoestq — COMMENTED** [review]
actually we need the dataset info

File: src/datasets/builder.py (hunk 2 above)
Comment (polinaeterna) [reply src/datasets/builder.py:885]: hm true...
what should we do then? I don't think it's fine that we get different results running the same command, like:
```python
load_dataset("ds_name")  # error - failed verifications
# then
load_dataset("ds_name", verification_mode="no_checks")  # successfully loaded
# then
load_dataset("ds_name")  # successfully loaded despite it was an error before
```
am I missing something?

Comment (lhoestq) [reply src/datasets/builder.py:885]: I don't think we can do anything tbh - e.g. reloading a cached dataset works even if you don't have the downloaded raw files anymore so we can't check for their checksums

**Review by lhoestq — COMMENTED** [review]
Yea the fix for the values is definitely welcome, thank you :)

File: src/datasets/builder.py [hunk 3]
```diff
@@ -358,8 +358,10 @@ def __init__(
...
             with FileLock(lock_path):
                 if os.path.exists(self._cache_dir):  # check if data exist
-                    if len(os.listdir(self._cache_dir)) > 0:
-                        logger.info("Overwrite dataset info from restored data version.")
+                    if len(os.listdir(self._cache_dir)) > 0 and os.path.exists(
+                        path_join(self._cache_dir, config.DATASET_INFO_FILENAME)
+                    ):
+                        logger.info("Overwrite dataset info from restored data version if exists.")
                         self.info = DatasetInfo.from_directory(self._cache_dir)
                     else:  # dir exists but no data, remove the empty dir as data aren't available anymore
```
[the remark this answers is not in the document]
Comment (polinaeterna) [reply src/datasets/builder.py:365]: thanks for noticing, fixed it, can I merge?

**Review by lhoestq — APPROVED** [review]
Thanks :)

**Review by mariosasko — APPROVED** [review]
Good catch!

**Pull request merged**